File: done_with/done_with/report/offers_from_customers/offers_from_customers.js
// Copyright (c) 2025, Light and contributors
// For license information, please see license.txt

frappe.query_reports["Offers From Customers"] = {
	"filters": [
		{
			"fieldname": "customer_name",
			"label": __("Customer Name"),
			"fieldtype": "Link",
			"options": "Customer",
			"width":100,
			"reqd": 0
		},
		{
			"fieldname": "item_name",
			"label": __("Item Name"),
			"fieldtype": "Link",
			"options": "Item Sample",
			"width":100,
			"reqd": 0
		},
		{
			"fieldname": "price",
			"label": __("Sale Price"),
			"fieldtype": "Currency",
			"width":100,
			"reqd": 0
		},
		{
			"fieldname": "offer_price",
			"label": __("Offer Price"),
			"fieldtype": "Currency",
			"width":100,
			"reqd": 0
		}
	]	
};

File: done_with/done_with/report/offers_from_customers/offers_from_customers.py
# Copyright (c) 2025, Light and contributors
# For license information, please see license.txt

import frappe


def execute(filters=None):
    columns = get_columns()
    data = get_data(filters)
    return columns, data

def get_data(filters):
    conditions = ""

    if filters.get("customer_name"):
        conditions += f" AND c.customer_name='{filters.get('customer_name')}'"

    if filters.get("item_name"):
        conditions += f" AND i.item_name = '{filters.get('item_name')}'"

    SQL = f"""
        SELECT
            c.customer_name,
            i.item_name,
            l.price,
            lo.offer_price
        FROM
            `tabItem Sample` AS i
        JOIN
            `tabListing` AS l
            ON i.name = l.item_name
        JOIN
            `tabListing Request` AS lo
            ON lo.parent = l.name
        JOIN
            `tabCustomer` AS c
            ON c.name = lo.full_name
        WHERE
            1=1 {conditions}
        ORDER BY l.item_name
    """

    return frappe.db.sql(SQL, as_list=True)


def get_columns():
	return [
		"Customer Name: Link/Customer: 200",
		"Item: Link/Item Sample: 200",
		"Sale Price: Currency:150",
		"Offer Price: Currency: 150"
	]

	
